### ISSUE TITLE
Converting circular structure to JSON

### ISSUE THREAD

**Issue by pselis** [ngx-formly/ngx-formly#1426]
```
[ ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ X] support request
```

Hi,
I'm using formly inside a "Matdialog", and I found something strange.
When I open the dialog for the second time the mandatory fields are already "touched",
and if i try to stringify the form fields (the second time) I get this error 
ERROR
Error: Converting circular structure to JSON

I can not understand where I'm wrong.

**Minimal reproduction of the problem with instructions**
https://stackblitz.com/edit/angular-3rd8sk-ytzwep

**Comment by aitboudad** [comment]
passed before it gets initialized by `formly-form` component:

**Comment by aitboudad** [comment]
cloning the `fields` before it gets initialized by `formly-form` component would solve the issue:
```ts
  constructor(@Inject(MAT_DIALOG_DATA) public data: any) { 
    this.model = data.item;
    this.fields = <FormlyFieldConfig[]>JSON.parse(JSON.stringify(data.fields));
  }
```

**Comment by aitboudad** [comment]
Well, I'm looking for a fix since I think this can be solved from our side.

**Comment by aitboudad** [comment]
try with `5.0.0-rc.4` which solve the issue, you don't have to use `JSON.stringify` either.

**Comment by pselis** [comment]
Thank you very much!

**Comment by aitboudad** [comment]
this fixed was reverted in `5.0.0-rc.5` in favor of https://github.com/ngx-formly/ngx-formly/issues/1429#issuecomment-465186967
